### ISSUE TITLE
ioctl: Test also struct termios in ioctl tests

### ISSUE THREAD

**Issue by pevik** [linux-test-project/ltp#637]
ATM we're testing just legacy `struct termio` in ioctl01.c and ioctl02.c.

https://lists.linux.it/pipermail/ltp/2020-January/015236.html
https://patchwork.ozlabs.org/patch/1230659/

**Comment by coolgw** [comment]
https://patchwork.ozlabs.org/project/ltp/list/?series=339112

**Comment by Martchus** [comment]
The patchwork link shows no results but it looks like this has already been implemented via https://github.com/linux-test-project/ltp/commit/813457d01c2b3db6e1f5f1fa4291b932e132bb30. So this issue can probably be closed.

**Comment by metan-ucw** [comment]
Looking at the patch https://github.com/linux-test-project/ltp/commit/813457d01c2b3db6e1f5f1fa4291b932e132bb30 it seems to be wrong though. The TCGETA for termio is called TCGETS for termios, we need separate lines with different ioctl cmds for different structures. And looking at man ioctl_tty there is termios2 too, with TCGETS2, hence we need to cover that as well.

**Comment by Martchus** [comment]
Ok, then maybe a ticket for me to pick-up after all. I guess I'll have a look whether I can make sense of what you're saying.

**Comment by Martchus** [comment]
I think I understood it. I'll send a patch for the first part (TCGETA vs. TCGETS). It is a while ago since I have sent a patch to a mailing list so likely this will be more complicated than the change itself.

**Comment by Martchus** [comment]
Yes. I'll work on ioctl02 tomorrow. Cleaning it up should be a good way of getting familiar with the code base.

**Comment by Martchus** [comment]
Note that I've send patches to the LTP list for `ioctl01.c` and the refactoring part of `ioctl02.c`. I'm currently waiting for feedback.

**Comment by Martchus** [comment]
The improvements for `ioctl01.c` have been merged so termios is tested (using the correct request). I'm still waiting for feedback for the refactoring of `ioctl02.c`.

**Comment by pevik** [comment]
> The patchwork link shows no results but it looks like this has already been implemented via [813457d](https://github.com/linux-test-project/ltp/commit/813457d01c2b3db6e1f5f1fa4291b932e132bb30).

FYI: When you click to icon next to _Action Required_ to disable this filter, you see the patch with status _Changes Requested_, which is disabled by default view:
https://patchwork.ozlabs.org/project/ltp/list/?series=339112&state=*
